addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/converters/OnOffTypeWSIntegerValueConverter.java

```diff
@@ -33,7 +37,10 @@ public OnOffType convertFromResourceValue(@NonNull WSIntegerValue from,
     @Override
     public WSIntegerValue convertFromOHType(@NonNull OnOffType from, @NonNull WSIntegerValue value,
             @NonNull ConverterAdditionalInfo convertData) throws ConversionException {
-        int newVal = from == OnOffType.ON ? value.maximumValue : value.minimumValue;
+
+        int onLevel = Collections
+                .min(Arrays.asList(value.maximumValue, getCommandLevel(value, convertData, OnOffType.ON)));
```

Comment:
This looks like a complex way to say `Math.min`?

Reply:
Ha, there was initially more than two values :blush: